BlazarService/src/main/java/com/hubspot/blazar/resources/InstantMessageResource.java

```diff
@@ -16,32 +12,23 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import com.google.common.base.Optional;
 import com.google.inject.Inject;
 import com.hubspot.blazar.base.notifications.InstantMessageConfiguration;
 import com.hubspot.blazar.data.service.InstantMessageConfigurationService;
-import com.hubspot.blazar.externalservice.slack.SlackChannel;
 import com.hubspot.jackson.jaxrs.PropertyFiltering;
-import com.ullink.slack.simpleslackapi.SlackSession;
 
 @Path("/instant-message")
 @Produces(MediaType.APPLICATION_JSON)
 @Consumes(MediaType.APPLICATION_JSON)
 public class InstantMessageResource {
 
-  private static final Logger LOG = LoggerFactory.getLogger(InstantMessageResource.class);
-  @Inject(optional = true)
+  @Inject
   private final InstantMessageConfigurationService instantMessageConfigurationService;
-  private Optional<SlackSession> slackSession;
 
   @Inject
-  public InstantMessageResource(InstantMessageConfigurationService instantMessageConfigurationService,
-                                Optional<SlackSession> slackSession) {
+  public InstantMessageResource(InstantMessageConfigurationService instantMessageConfigurationService) {
```

Comment:
Make it more like a restful resource:
@Path("/instant-message-configurations")

Name the class: InstantMessageConfigurationResource()
and then @GET, @POST don't need a @Path (it is @Path("/")) and PUT is @PUT(/{id}), etc.